consensus/common/src/main/java/tech/pegasys/pantheon/consensus/common/ValidatorVote.java

```diff
@@ -12,8 +12,62 @@
  */
 package tech.pegasys.pantheon.consensus.common;
 
-public interface ValidatorVote {
-  boolean isAddVote();
+import static tech.pegasys.pantheon.consensus.common.VoteType.ADD;
 
-  boolean isDropVote();
+import tech.pegasys.pantheon.ethereum.core.Address;
+
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+
+public class ValidatorVote {
+
+  private final VoteType votePolarity;
+  private final Address proposer;
+  private final Address recipient;
+
+  public ValidatorVote(
+      final VoteType votePolarity, final Address proposer, final Address recipient) {
+    Preconditions.checkNotNull(votePolarity);
+    Preconditions.checkNotNull(proposer);
+    Preconditions.checkNotNull(recipient);
```

Comment:
nit: Can we static import `checkNotNull` please?  Just to keep things succinct.

Reply:
done.

Reply:
Done.